homeassistant/components/smlight/sensor.py

```diff
@@ -0,0 +1,166 @@
+        value_fn=lambda x: x.sensors.esp32_temp,
+    ),
+    SmSensorEntityDescription(
+        key="zb_temperature",
+        translation_key="zigbee_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value_fn=lambda x: x.sensors.zb_temp,
+    ),
+    SmSensorEntityDescription(
+        key="core_uptime",
+        translation_key="core_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.uptime,
+    ),
+    SmSensorEntityDescription(
+        key="socket_uptime",
+        translation_key="socket_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.socket_uptime,
+    ),
+    SmSensorEntityDescription(
+        key="ram_usage",
+        translation_key="ram_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.ram_usage,
+    ),
+    SmSensorEntityDescription(
+        key="fs_usage",
+        translation_key="fs_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.fs_used,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SMLIGHT sensor based on a config entry."""
+    coordinator: SmDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [SmSensorEntity(coordinator, description) for description in SENSORS]
+    async_add_entities(sensors)
+
+
+class SmSensorEntity(
+    SmEntity, CoordinatorEntity[SmDataUpdateCoordinator], SensorEntity
+):
+    """Representation of a slzb sensor."""
+
+    entity_description: SmSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SmDataUpdateCoordinator,
+        description: SmSensorEntityDescription,
+    ) -> None:
+        """Initiate slzb sensor."""
+        super().__init__(coordinator)
+        CoordinatorEntity.__init__(self, coordinator)
+
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.unique_id}_{description.key}"
+        self._last_uptime: datetime | None = None
+
+    def get_device_uptime(self, uptime: float | None) -> datetime:
```

Comment:
I would move the whole uptime stuff to a separate sensor class that is independent from the standard sensors, would make the code clearer to read imo

Reply:
yes agree that is probably clearer